proto/src/determined/api/v1/master.proto

```diff
@@ -116,6 +135,32 @@ message PatchMasterConfigResponse {
   google.protobuf.Struct config = 1;
 }
 
+// Post new maintenance messages.
+message PostMaintenanceMessageRequest {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "message", "start_time", "end_time" ] }
+  };
+  // Text content of message.
+  string message = 1;
+  // Time to begin showing message.
+  google.protobuf.Timestamp start_time = 2;
+  // Time to stop showing message.
+  google.protobuf.Timestamp end_time = 3;
```

Comment:
```suggestion
  optional google.protobuf.Timestamp end_time = 3;
```

Since `end_time` is optional, perhaps we should mark this field with `optional` for clarity.